### PR TITLE
Kevinsolorio/rep 16 viewing claiming pickup tasks display available tasks

### PR DESCRIPTION
**Pull request by kevin3656**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
This added the ability to retrieve tasks from the replate database. Changes were made in replate-business to add this endpoint and then in this repo to fetch these tasks.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
IMG_1424

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
In the branch with same name from replate-business:

generate the demo data with bin/rails dev:gen_today_tasks N=5
run bin/rails s -b 0.0.0.0
navigate to api/tasks endpoints, should see json with a list of todays pickups(mock data)

Once you verify the tasks are being created move to this repo(the blueprint one, from this pull request):
change "EXPO_PUBLIC_BACKEND_URL": "http:10.40.164.190:3000" on line 30 from app.json to your ip adress
do the same for line 15 of src/app/available-pick-ups/index.tsx

now run app and the tasks should display wooo

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
populating the details once you tap on them, and cleaning up the date/time formatting.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @insert pl github username here

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -25,6 +25,9 @@
     "web": {
       "favicon": "./assets/bp-favicon.png"
     },
-    "plugins": ["expo-router"]
+    "plugins": ["expo-router"],
+    "extra": {
+      "EXPO_PUBLIC_BACKEND_URL": "http:10.40.164.190:3000"
```
Comment (stephenhungg) [inline app.json]: Should be http://10.40.164.190:3000, might give invalid url otherwise no?

File: src/app/available-pick-ups/index.tsx [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+type AppExtra = { EXPO_PUBLIC_BACKEND_URL?: string };
+
+const rawExtra: unknown = Constants.expoConfig?.extra;
+const extra: AppExtra =
+  rawExtra && typeof rawExtra === 'object' ? (rawExtra as AppExtra) : {};
+
+export const BACKEND_URL =
+  typeof extra.EXPO_PUBLIC_BACKEND_URL === 'string'
+    ? extra.EXPO_PUBLIC_BACKEND_URL
+    : '10.40.164.190:3000';
```
Comment (stephenhungg) [inline src/app/available-pick-ups/index.tsx]: same issue, might need to change to http://10.40.164.190:3000

File: src/app/pickup-details/[id].tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    >
+      <Text style={{ fontWeight: '700', marginBottom: 6 }}>{title}</Text>
+      {children}
+    </View>
+  );
+}
+
+export default function PickupDetails() {
+  const { id } = useLocalSearchParams<{ id: string }>();
+  const pickup = MOCK_PICKUPS.find(p => String(p.id) === String(id));
```
Comment (stephenhungg) [inline src/app/pickup-details/[id].tsx:25]: Right now the screen looks up tasks in MOCK_PICKUPS, and this is probably going to fail whenever you use real API results. Update navigation to pass task data, so like when you call router.push, include { params: {...item} }.
In the detail screen, read the params with useLocalSearchParams and use those values instead of MOCK_PICKUPS.find. I haven't actually fully tested out the implementation cuz I don't think your changes are merged on replate-business but if it works don't change this lol.

**Review by stephenhungg — COMMENTED** [review]
Overall should have no issues except the find by mock data but looks good!

File: src/app/available-pick-ups/index.tsx [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+      } catch (e: unknown) {
+        setError(getErrorMessage(e));
+      }
+    })();
+  }, []);
+  const days = React.useMemo(() => {
+    const out: string[] = [];
+    const start = new Date();
+    start.setDate(start.getDate() - 1);
+    for (let i = 0; i < 5; i++) {
```
Comment (rachaelch3n) [inline src/app/available-pick-ups/index.tsx]: should only be rendering 2 days -- we fixed this on the last pr, did u rebase?
